### PR TITLE
feat: [CDS-47126]: Force deletion of Environment Entity in NG

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
Adding a query param for force delete in case of Environment
Force deletion of Environment Entity in NG as per https://harness.atlassian.net/wiki/spaces/PLATFORM/pages/21237040305/Force+Delete+support+for+Harness+Entities
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)

### CODE REVIEW

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java [hunk 1]
```diff
@@ -114,6 +124,8 @@ public class EnvironmentServiceImpl implements EnvironmentService {
   private final ClusterService clusterService;
   private final ServiceOverrideService serviceOverrideService;
   private final ServiceEntityService serviceEntityService;
+  @Inject private AccountClient accountClient;
+  @Inject NGSettingsClient settingsClient;
```
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java:128]: nit: can we also inject these 2 through constructor to maintain consistency?

**Review by yogesh-chauhan — APPROVED** [review]

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 7 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)* status: 1 file(s) to review.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)* status: :shipit: complete! ✓

**Comment by vivekkarnatiharness** [comment]
trigger ti0

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)* status: 12 file(s) to review. 1 of 2 approvals obtained.

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Review by RajBaviskar — APPROVED** [review]

**Review by RajBaviskar — APPROVED** [review]
> Reviewed 6 of 8 files at r1, 1 of 1 files at r3, 3 of 3 files at r4, 2 of 2 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42627)* status: :shipit: complete! ✓

**Pull request merged**